lib/ansible/modules/network/aci/aci_aaa_user.py

```diff
@@ -0,0 +1,239 @@
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: aci
+'''
+
+EXAMPLES = r'''
+- name: Add a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    aaa_password: AnotherSecretPassword
+    expiration: never
+    expires: no
+    email: dag@wieers.com
+    phone: 1-234-555-678
+    first_name: Dag
+    last_name: Wieers
+    state: present
+
+- name: Remove a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    state: absent
+
+- name: Query a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    state: query
+
+- name: Query all users
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    state: query
+'''
+
+RETURN = r''' # '''
+
+from ansible.module_utils.network.aci.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        aaa_password=dict(type='str', no_log=True),
+        aaa_password_lifetime=dict(type='int'),
+        aaa_password_update_required=dict(type='bool'),
+        aaa_user=dict(type='str', required=True, aliases=['name']),
+        clear_password_history=dict(type='bool'),
+        description=dict(type='str', aliases=['descr']),
+        email=dict(type='str'),
+        enabled=dict(type='bool'),
+        expiration=dict(type='str'),
+        expires=dict(type='bool'),
+        first_name=dict(type='str'),
+        last_name=dict(type='str'),
+        phone=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['aaa_user']],
+            ['state', 'present', ['aaa_user']],
+            ['expires', True, ['expiration']],
+        ],
+    )
+
+    aaa_password = module.params['aaa_password']
+    aaa_password_lifetime = module.params['aaa_password_lifetime']
+    aaa_password_update_required = module.params['aaa_password_update_required']
+    aaa_user = module.params['aaa_user']
+    clear_password_history = module.params['clear_password_history']
+    description = module.params['description']
+    email = module.params['email']
+    enabled = module.params['enabled']
+    expiration = module.params['expiration']
+    first_name = module.params['first_name']
+    last_name = module.params['last_name']
+    phone = module.params['phone']
+    state = module.params['state']
+
+    if module.params['enabled'] is True:
+        enabled = 'active'
+    elif module.params['enabled'] is False:
+        enabled = 'inactive'
+    else:
```

Comment:
Ansible defaults unpassed arguments to `None`, and checks that the values entered are `True` or `False` since type is 'bool', so there is no need to have `else: None`

Reply:
We still have to pass the value over to variable `enabled`.

Reply:
Yes, that is not the point I was trying to make. Something like this should account for that:
```
if expiration is never:
    expires = False
elif expiration is not None:
    expires = True
```
If expiration is not passed, then expires would also not be passed and let the APIC handle default behavior for new accounts

Reply:
ya, I would set enabled = module.params['enabled'] and then do that, as this only requires dictionary access once (not really much of a performance difference with dict lookups, but still 1 > 2 :D )

Reply:
But in this case you do 2 assignments instead of one, except when it is None ;-)
Oh well...